ceph/ceph.go

```diff
@@ -178,148 +281,173 @@ func matchSlice(a, b []string) bool {
 	}
 
 	for i := range a {
-		if a[i] != b[i] && a[i] != "*" && b[i] != "*" {
+		if !matchEntry(a[i], b[i]) {
 			return false
 		}
 	}
 
 	return true
 }
 
-func (ceph *Ceph) getJSONDataByNamespace(data map[string]interface{}, namespace []string, resultNamespace []string, results map[string]interface{}) {
+// getJSONPathsByNamespace returns all JSON path - value entries for provided namespace
+// If provided namespace is empty, all JSON paths are returned
+// If dataType is jsonSchema, schema data will be returned as path -> map[description, type, nick] instead of separate path -> value entries
+// If dataType is JSON_DATA, every JSON data value will be returned as separate path -> value entry
+func (ceph *Ceph) getJSONPathsByNamespace(data map[string]interface{}, dataType int, namespace []string, resultNamespace []string, results map[string]interface{}) {
 	// Go through all JSON data keys
-	for key, _ := range data {
+	for key := range data {
 
-		// Convert ceph key to namespace slice for comparsion
-		keyNs := strings.Split(key, ".")
+		// Convert ceph key to namespace slice for comparsion, for ex. osd/1/metric.0/wait will be split into osd/1/metric/0/wait
+		//keyNs := strings.Split(key, ".")
```

Comment:
No, uncommenting this just adds support for splitting namespace elements which have "." in name, but i removed that feature. I'll remove that comment.